pkg/apiserver/controller/queue/queue.go

```diff
@@ -338,26 +338,33 @@ func UpdateQueue(ctx *logger.RequestContext, request *UpdateQueueRequest) (Updat
 
 	// validate fields if not nil, validate namespace at first
 	updateClusterRequired := false
+	resourceUpdated := true
 
 	// validate MaxResource or MinResource
-	scalarResourceLaws := config.GlobalServerConfig.Job.ScalarResourceArray
-	if request.MaxResources.CPU != "" && request.MaxResources.Mem != "" {
-		if err = schema.ValidateResourceInfo(request.MaxResources, scalarResourceLaws); err != nil {
-			ctx.Logging().Errorf("update queue failed. error: %s", err.Error())
+	if request.MinResources.CPU != "" || request.MinResources.Mem != "" {
+		if err := validateQueueResource(request.MaxResources, &queueInfo.MaxResources); err != nil {
+			ctx.Logging().Errorf("update queue maxResources failed. error: %s", err.Error())
 			ctx.ErrorCode = common.InvalidComputeResource
 			return UpdateQueueResponse{}, err
 		}
 		updateClusterRequired = true
-		queueInfo.MaxResources = request.MaxResources
+		resourceUpdated = true
 	}
-	if request.QuotaType == schema.TypeElasticQuota && request.MinResources.CPU != "" && request.MinResources.Mem != "" {
-		if err = schema.ValidateResourceInfo(request.MinResources, scalarResourceLaws); err != nil {
-			ctx.Logging().Errorf("update queue failed. error: %s", err.Error())
+	if queueInfo.QuotaType == schema.TypeElasticQuota && (request.MinResources.CPU != "" || request.MinResources.Mem != "") {
+		if err := validateQueueResource(request.MinResources, &queueInfo.MinResources); err != nil {
+			ctx.Logging().Errorf("update queue minResources failed. error: %s", err.Error())
 			ctx.ErrorCode = common.InvalidComputeResource
 			return UpdateQueueResponse{}, err
 		}
 		updateClusterRequired = true
-		queueInfo.MinResources = request.MinResources
+		resourceUpdated = true
+	}
+	if queueInfo.QuotaType == schema.TypeElasticQuota && resourceUpdated &&
+		!request.MinResources.LessEqual(request.MaxResources) {
```

Comment:
If we only update `MinResources` and `MaxResources` is empty, this check might be wrong.

Reply:
agree it